src/components/Network.js

```diff
@@ -33,27 +33,49 @@ class Network extends React.Component {
       connected_peers: [],
       known_peers: [],
       loaded: false,
-      isLoading: false
+      isLoading: false,
+      peers: [],
+      peerId: '',
     }
 
-    this.loadData = this.loadData.bind(this);
   }
 
   componentDidMount() {
-    this.loadData();
-    this.loadTimer = setInterval(this.loadData, 1000);
+    this.loadPeers();
+  }
+
+  loadPeers() {
+    const { match: { params } } = this.props;
+
+    networkApi.getPeerList().then((peers) => {
+      this.setState({peers}, () => {
+        let peerId = peers.find(target => target === params.peerId);
+        if (!peerId) {
+          peerId = peers[0];
```

Comment:
Should we handle the case when the API returns an empty array of peers? It should throw an error here, right? I know it shouldn't happen but I am thinking if we should do something to protect this. What do you think?

Reply:
I'm not sure where is my response for @msbrogli about what happens when the request fail. 

What happen is: the Loading will be shown until page reload. As we don't have a good way to show errors nowadays, i think we can wait for refactors that will be made soon.

Reply:
I was not talking about when a request fails. I was saying that the request may succeed and returns an empty array. You are assuming the array will always have at least one element and doing `peers[0]`. As we don't have a good way to show errors, you can at least write a TODO (Check if array is empty and show error)

Reply:
It's not exactly the same thing but is the same behavior: with an empty list `peers[0]` will result in `undefined`, so, the next request will pass `undefined` as id and the request will fail. The best we can do here, for now, is try to fetch node ids again instead of just go ahead. But the behavior will remain the same, except for the very edge case when the peer list is empty when user load the page and start to have items while the page is open.